akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2AlpnSupport.scala

```diff
@@ -32,48 +29,31 @@ private[http] object Http2AlpnSupport {
    */
   def enableForServer(engine: SSLEngine, setChosenProtocol: String => Unit): SSLEngine =
     if (isAlpnSupportedByJDK) jdkAlpnSupport(engine, setChosenProtocol)
-    else jettyAlpnSupport(engine, setChosenProtocol)
+    else throw new RuntimeException(s"Need to run on a JVM >= 8u252 for ALPN support needed for HTTP/2. Running on ${sys.props("java.version")}")
 
   def isAlpnSupportedByJDK: Boolean =
     // ALPN is supported starting with JDK 9
-    JavaVersion.majorVersion >= 9
+    JavaVersion.majorVersion >= 9 ||
+      (classOf[SSLEngine].getMethods.exists(_.getName == "setHandshakeApplicationProtocolSelector")
```

Comment:
This will probably not help anymore because class loading will now fail on older JDKs. Should we keep the reflective access below to avoid that? Or create another class, so we can have a reflective check before the actual one?

Reply:
Just leaving reflection seems enough to me.

Reply:
Referring to missing methods does not prevent class loading, only referring to unknown types does. For a missing method the class loads fine, but a `java.lang.NoSuchMethodError` will be thrown when we try to invoke the method.

So this seems :+1: